ODPi-Runtime.md

```diff
@@ -233,6 +233,19 @@ The following requirements provide guidance around best practices of integrating
 * **[HCFS_INTEGRATION1]** First piece of Matt's doc goes here
 * **[HCFS_INTEGRATION2]** Second piece of Matt's doc goes here
 
+-   Applications SHOULD define APIs and behaviors explicitly in the FileSystem class instead of the HDFS class. A file system will be HCFS compatible if all the APIs are implemented from the FileSystem class. Defining API specification in the HDFS class is considered optional usage in HCFS distributed file system.
+
+-   ODPi Platforms MAY define optional storage tiers functions that can be implemented by different solutions. 
+
+-   ODPi Platforms SHOULD provide error handling for applications invoking HDFS specific APIs that are not included in the FileSystem Class.
+
+-   ODPi Platforms SHOULD be compatible with POSIX ACL and POSIX permissions.
```

Comment:
Do you wish to also note that when a new file is created in HDFS, its owner is of course the user id of the process that created it, but the group is inherited from the parent directory (not the user's primary group) -- and it is recommended that HCFS implementations follow this rule also, because HDFS-using applications expect it?  This means following the "BSD rule" instead of the "System V rule" for group ownership of new files.